File: src/main/java/by/architecture_map/belarus/entity/ArchitecturalAttribute.java
package by.architecture_map.belarus.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * Attributes that may be characteristic of the architectural
 * style, for example, pilasters, lancet windows, etc.
 */

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class ArchitecturalAttribute extends BaseEntity {

    private String name;
    private String description;

    @OneToOne
    @JoinColumn(name = "demonstrative_image_id", referencedColumnName = "id")
    private Image demonstrativeImage;
}

File: src/main/kotlin/by/architecture_map/belarus/entity/ArchitecturalAttribute.kt
package by.architecture_map.belarus.entity

import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne

/**
 * Attributes that may be characteristic of the architectural
 * style, for example, pilasters, lancet windows, etc.
 */

@Entity
data class ArchitecturalAttribute(

    var name: String? = null,
    var description: String? = null,
    @OneToOne
    @JoinColumn(name = "demonstrative_image_id", referencedColumnName = "id")
    var demonstrativeImage: Image? = null

) : BaseEntity()
